packages/vm/src/parsed_wasm.rs

```diff
@@ -1,10 +1,40 @@
+use std::{fmt, mem};
+
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
 use wasmer::wasmparser::{
-    CompositeType, Export, Import, MemoryType, Parser, Payload, TableType, ValidPayload, Validator,
-    WasmFeatures,
+    BinaryReaderError, CompositeType, Export, FuncToValidate, FunctionBody, Import, MemoryType,
+    Parser, Payload, TableType, ValidPayload, Validator, ValidatorResources, WasmFeatures,
 };
 
 use crate::{VmError, VmResult};
 
+#[derive(Default)]
+pub struct OpaqueDebug<T>(pub T);
```

Comment:
Could you add a short doc comment why this type exists and what it does?

Reply:
Added some docs. Essentially it's needed to implement `Debug` on the struct because `wasmparser::FuncToValidate` doesn't implement it and just removing a `Debug` derive because of this one member felt wrong.

Reply:
Nice, thanks